SeaSharpe_CVGS/SeaSharpe_CVGS/Models/ValidateAttributes.cs

```diff
@@ -0,0 +1,115 @@
+ * 
+ */
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Web;
+
+namespace SeaSharpe_CVGS.Models
+{
+    public class ValidateAttributes
+    {
+        //will hold the temporary value of startDate for an event. 
+        //Use it to compare it with end date of an event
+        public static DateTime EventStartDate;
+    }
+
+    /// <summary>
+    /// Will make sure the value pass is not in the past 
+    /// 
+    /// Use it by: Event Entity
+    /// </summary>
+    public class DateNotInThePast : ValidationAttribute
+    {
+        public override bool IsValid(object value)
+        {
+            DateTime dt;
+            bool parsed = DateTime.TryParse((string)value.ToString(), out dt);
+            if (!parsed)
+            {
+                return false;
+            }
+
+            if (dt < DateTime.Today)
+            {
+                return false;
+            }
+
+            ValidateAttributes.EventStartDate = dt;
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Will make sure End Date of an event is later than the Start Date (ValidateAction.EventStartDate)
+    /// 
+    /// Use it by: Event Entity
+    /// </summary>
+    public class DateNotBeforeStartDate : ValidationAttribute
+    {
+        public override bool IsValid(object value)
+        {
+            var db = new ApplicationDbContext();
+
+            DateTime dt;
+            bool parsed = DateTime.TryParse((string)value.ToString(), out dt);
+            if (!parsed)
+            {
+                return false;
+            }
+
+            if (dt < ValidateAttributes.EventStartDate)
+            {
+                return false;
+            }
+
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Evaluates if object passed match the regular expresion
+    /// 
+    /// Use it by: Address Entity
+    /// </summary>
+    public class PostalCodeValidation : ValidationAttribute
+    {
+        /// <summary>
+        /// Evaluates if object passed match the regular expresion
+        /// </summary>
+        /// <param name="value">Object to be evaluated</param>
+        /// <returns>True if the postal code is null or whitespaces, 
+        /// True if the regular expression match the object value, false otherwise</returns>
+        public override bool IsValid(object value)
+        {
+            string postalCode = "";
+            if (value != null)
+            {
+                postalCode = value as string;
+                postalCode = postalCode.Trim();
+
+                if (string.IsNullOrWhiteSpace(postalCode))
+                {
+                    return true;
+                }
+                else
+                {
+                    return Regex.IsMatch
+                        (postalCode, @"^(?i)[ABCEGHJKLMNPRSTVXY][0-9][ABCEGHJKLMNPRSTVWXYZ] ?[0-9][ABCEGHJKLMNPRSTVWXYZ][0-9]$");
```

Comment:
This regex says that both `H0H 0H0` and `H0H0H0`, I agree that we shouldn't give an error for both of these but we should probably make sure we are storing the postal code the same way regardless of how the user enters it. 

Until we add logic to ensure that postal codes are converted to have a space (or not to have a space) it would be a good idea for the regex to be more strict.